src/mlpack/methods/ann/layer/elu_impl.hpp

```diff
@@ -51,12 +51,27 @@ template<typename InputType, typename OutputType>
 void ELU<InputDataType, OutputDataType>::Forward(
     const InputType&& input, OutputType&& output)
 {
-  Fn(input, output);
-
-  if (!deterministic)
+  output.set_size(arma::size(input));
+  for (size_t i = 0; i < input.n_elem; i++)
```

Comment:
I think we should avoid use of for loops and use armadillo methods as using GPU can be faster. I can be wrong.

Reply:
I think that should be fine, I have benchmarking on my to do list (already had a PR open for the same), Once the same is established I will make changes for all activation / loss functions. Currently this is just a simple restructuring to make the code a bit more consistent. Thanks.

Reply:
Hmmm, Should I make the changes now?
Well, I don't think I can resist taking a much closer look now. I will run some tests again and hopefully we can change everything to `arma` functions.

Reply:
I would be happy to collaborate in this task. After all there are a lot of files to be restructured

Reply:
> Hmmm, Should I make the changes now?

Don't think it should be part of the PR.

Reply:
Agreed, So I'll keep it as it is for now, will reopen my previous PR after getting fixed benchmarks.
@prince776, I'll tag you in the PR once I reopen it and we can work on it.

Reply:
looking forward to it.